app/assets/javascripts/discourse/app/lib/push-notifications.js

```diff
@@ -132,5 +134,6 @@ export function unsubscribe(user, callback) {
     if (callback) {
       callback();
     }
+    return true;
```

Comment:
I think you might need to return the `navigator.serviceWorker.ready` promise for this value not be thrown away

Reply:
great catch - pushed